### PR TITLE
ADEN-3484 Fandom UAP tests

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsFandomObject.java [hunk 1]
```diff
@@ -37,12 +37,31 @@ public AdsFandomObject(WebDriver driver, String testedPage) {
...
+  public long getLineItemId(String slotName) {
+    JavascriptExecutor js = driver;
+    try {
+      return (long) js.executeScript(
+          "var slots = googletag.getSlots(); for (var i = 0; i < slots.length; i++) { " +
+          "if (slots[i].getTargeting('pos').indexOf('" + slotName + "') !== -1) { " +
+          "return slots[i].getResponseInformation().lineItemId;" +
+          "} }"
+      );
+    } catch (WebDriverException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsFandomObject.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by jacekjursza** [comment]
LGTM

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsFandomObject.java [hunk 2]
```diff
@@ -40,9 +43,30 @@ public AdsFandomObject(WebDriver driver, String testedPage) {
...
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsFandomObject.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsFandomObject.java [hunk 3]
```diff
@@ -40,9 +43,30 @@ public AdsFandomObject(WebDriver driver, String testedPage) {
     slots.put(AdsFandomContent.BOTTOM_BOXAD, bottomBoxadElement);
   }
 
+  public void triggerOnScrollSlots() {
```
Comment (nandy-andy) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsFandomObject.java]: [JavascriptActions.scrollToBottom()](https://github.com/Wikia/selenium-tests/blob/39ba40131e196d76e27cae85aa8299f697c6b992/src/test/java/com/wikia/webdriver/common/core/elemnt/JavascriptActions.java#L77)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsFandomObject.java [hunk 4]
```diff
@@ -40,9 +43,30 @@ public AdsFandomObject(WebDriver driver, String testedPage) {
...
+    JavascriptExecutor js = driver;
+    js.executeScript("window.scrollTo(0, document.body.scrollHeight);");
+  }
+
+  public long getLineItemId(String slotName) {
```
Comment (nandy-andy) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsFandomObject.java]: Do you think `AdsFandomObject` is the best possible place for this method?

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsFandomObject.java [hunk 5]
```diff
@@ -3,6 +3,7 @@
 import com.wikia.webdriver.common.contentpatterns.AdsFandomContent;
 
 import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsFandomObject.java]: MINOR Remove this unused import 'org.openqa.selenium.JavascriptExecutor'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Pull request merged**